backend/src/main/java/ai/verta/modeldb/cron_jobs/DeleteEntitiesCron.java

```diff
@@ -0,0 +1,666 @@
+        if (ownerRoleBindingName != null) {
+          roleBindingNames.add(ownerRoleBindingName);
+        }
+      }
+
+      if (!roleBindingNames.isEmpty()) {
+        roleService.deleteRoleBindings(roleBindingNames);
+      }
+
+      Transaction transaction = session.beginTransaction();
+      String updateDeletedStatusExperimentRunQueryString =
+          new StringBuilder("UPDATE ")
+              .append(ExperimentRunEntity.class.getSimpleName())
+              .append(" expr ")
+              .append("SET expr.")
+              .append(ModelDBConstants.DELETED)
+              .append(" = :deleted ")
+              .append(" WHERE expr.")
+              .append(ModelDBConstants.EXPERIMENT_ID)
+              .append(" IN (:experimentIds)")
+              .toString();
+      Query deletedExperimentRunQuery =
+          session.createQuery(updateDeletedStatusExperimentRunQueryString);
+      deletedExperimentRunQuery.setParameter("deleted", true);
+      deletedExperimentRunQuery.setParameter("experimentIds", experimentIds);
+      deletedExperimentRunQuery.executeUpdate();
+
+      for (ExperimentEntity experimentEntity : experimentEntities) {
+        session.delete(experimentEntity);
+      }
+      transaction.commit();
+    }
+
+    LOGGER.debug(
+        "Experiment deleted successfully : Deleted experiments count {}", experimentIds.size());
+  }
+
+  private void deleteExperimentRuns(Session session) {
+    LOGGER.debug("ExperimentRun deleting");
+    String deleteExperimentRunQueryString =
+        new StringBuilder("FROM ")
+            .append(ExperimentRunEntity.class.getSimpleName())
+            .append(" expr WHERE expr.")
+            .append(ModelDBConstants.DELETED)
+            .append(" = :deleted ")
+            .toString();
+
+    Query experimentRunDeleteQuery = session.createQuery(deleteExperimentRunQueryString);
+    experimentRunDeleteQuery.setParameter("deleted", true);
+    experimentRunDeleteQuery.setMaxResults(this.recordUpdateLimit);
+    LOGGER.debug("ExperimentRun delete query: {}", experimentRunDeleteQuery.getQueryString());
+    List<ExperimentRunEntity> experimentRunEntities = experimentRunDeleteQuery.list();
+
+    List<String> experimentRunIds = new ArrayList<>();
+    if (!experimentRunEntities.isEmpty()) {
+      List<String> roleBindingNames = new LinkedList<>();
+      for (ExperimentRunEntity experimentRunEntity : experimentRunEntities) {
+        experimentRunIds.add(experimentRunEntity.getId());
+        String ownerRoleBindingName =
+            roleService.buildRoleBindingName(
+                ModelDBConstants.ROLE_EXPERIMENT_RUN_OWNER,
+                experimentRunEntity.getId(),
+                experimentRunEntity.getOwner(),
+                ModelResourceEnum.ModelDBServiceResourceTypes.EXPERIMENT_RUN.name());
+        if (ownerRoleBindingName != null) {
+          roleBindingNames.add(ownerRoleBindingName);
+        }
```

Comment:
Same.

Reply:
Created separate functions called `deleteROleBindingsFor*` and moved all delete role bindings code in that function for each entity.